geode/exact/circle_csg.cpp

```diff
@@ -249,6 +249,139 @@ static void random_circle_quantize_test(int seed) {
     GEODE_ASSERT(unioned.size() <= arcs.size());
   }
 }
+
+static void circle_arc_area_test() {
+  // These should return exactly zero with no rounding error
+  GEODE_ASSERT(circle_arc_area(Array<CircleArc>()) == 0.);
+  GEODE_ASSERT(circle_arc_area(Nested<CircleArc>()) == 0.);
+  const auto degenerate = make_array({CircleArc(Vec2(2.,2.),1.)});
+  // No csg functions should return an arc polygon with a single vertex, but circle_arc_area should be able to handle it
+  GEODE_ASSERT(circle_arc_area(degenerate) == 0.);
+
+  const auto area_matches = [](const real area, const real expected) -> bool {
+    constexpr real area_tolerance = 1e-9; // Maximum expected delta due to floating point rounding
+    return abs(area - expected) < area_tolerance;
+  };
+
+  // Check result for a square
+  auto unit_square = make_array({CircleArc(Vec2(0,0),0), CircleArc(Vec2(1,0),0), CircleArc(Vec2(1,1),0), CircleArc(Vec2(0,1),0)});
+  GEODE_ASSERT(area_matches(circle_arc_area(unit_square), 1.));
+  reverse_arcs(unit_square); // Should negate area
```

Comment:
I'm not sure I understand why -- mathematically speaking -- reversing a sequence of arcs would produce a negative area.